### PR TITLE
Print offsets in promql.

### LINKED ISSUE #1223: offset keyword not working in recorded rules or alert rules
Having problems with "offset" in recorded rules and alerts.

Full expression works in templates and expression browser, but when a recorded rule or alert rule, "offset" gets ignored.

For example, with these two rules.  Evaluating the two new timeseries in the browser, they always report the same number -- and on the status page rule list there's no mention of "offset"

```
net_total_rate = sum(rate(node_network_receive_bytes{device="eth0"}[5m])) * 8
net_total_rate_offset = sum(rate(node_network_receive_bytes{device="eth0"}[5m] offset 1m)) * 8
```

***

Screenshot of status page, note missing "offset" keyword on the right hand side:

prom_1_rules

***

Screenshot of first rule:

prom_2

Screenshot of offset rule, note that the value is reported as the same, it should not be:

prom_3

***

Evaluating the expression in the browser itself works, and the "offset" version does report the offset:

prom_4

prom_5

***

The above examples use "offset 1m", but the same issue persists with 5m, 30m, ... the recorded rules don't use the offset.
(fyi this test prometheus instance has scrape/evaluation intervals of 5s)

### PR DESCRIPTION
**Pull request by brian-brazil**
Fixes #1223

### CODE REVIEW

File: promql/printer.go [hunk 1]
```diff
@@ -174,7 +175,11 @@ func (node *MatrixSelector) String() string {
 		Name:          node.Name,
 		LabelMatchers: node.LabelMatchers,
 	}
-	return fmt.Sprintf("%s[%s]", vecSelector.String(), strutil.DurationToString(node.Range))
+	offset := ""
+	if node.Offset != time.Duration(0) {
+		offset = fmt.Sprintf(" offset %s", strutil.DurationToString(node.Offset))
```
Comment (juliusv) [inline promql/printer.go]: We capitalize all the other printed keywords. We should change them all to lower-case, but maybe better in a separate change. So here, maybe capitalize it for consistency?

**Comment by juliusv** [comment]
Ouch. This actually affects rule execution, as there the printed string is parsed back into an expression to execute the rule (something that would also be good to clean up at some point): https://github.com/prometheus/prometheus/blob/master/rules/alerting.go#L153

**Comment by juliusv** [comment]
:+1: otherwise

**Comment by juliusv** [comment]
(ah, seeing the related issue now, where the rule problem was mentioned)

File: promql/printer.go (hunk 1 above)
Comment (brian-brazil) [inline promql/printer.go]: Done

**Pull request merged**